store/schedule.go

```diff
@@ -350,14 +413,37 @@ func validateField(fieldData string, field string) string {
 	return ""
 }
 
-func validateScheduleTime(scheduleTime int64, appId string) string {
+func validateScheduleTime(scheduleTime int64, app App, maxTTL int) string {
 	now := time.Now().Unix()
 	if scheduleTime < now {
 		return fmt.Sprintf("schedule time : %d is less than current time: %d for app: %s. Time cannot be in past.",
 			scheduleTime,
 			now,
-			appId)
+			app.AppId)
+	} else if (scheduleTime - now) > int64(app.GetMaxTTL(maxTTL)) {
+		return fmt.Sprintf("schedule time : %d cannot be more than %d days from current time : %d for app: %s",
+			scheduleTime,
+			app.GetMaxTTL(maxTTL)/(24*60*60),
+			now,
+			app.AppId)
+	}
+	return ""
+}
+
+func validatePayloadSize(payload string, app App, maxPayload int) string {
+	var maxPayloadSize int
+
+	if app.Configuration.PayloadSize == 0 {
+		maxPayloadSize = maxPayload
+	} else {
+		maxPayloadSize = app.Configuration.PayloadSize
 	}
+
+	if len(payload) > maxPayloadSize {
+		glog.Errorf("PayloadSize for app: %s cannot be more than %d bytes, given payloadSize bytes: %d ", app.AppId, maxPayloadSize, len(payload))
```

Comment:
We should define a String variable which takes value as `fmt.Sprintf("PayloadSize for app: %s cannot be more than %d bytes, given payloadSize bytes: %d", app.AppId, maxPayloadSize, len(payload))` and pass that string into glog.Errorf and return the same string instead of defining it twise.

Reply:
Done.